applications/vanilla/models/class.discussionmodel.php

```diff
@@ -2464,4 +2464,38 @@ public static function allowedSortFields($Allowed = null) {
 
         return self::$AllowedSortFields;
     }
+
+
+    /**
+     * Tests whether a user has permission to view a specific discussion by checking category-specific permissions.
+     * Fires an event that can override the category view permission.
+     *
+     * @param object|array|integer $discussion The discussion ID or the discussion to test.
+     * @param string|integer $userID The ID of the user to test permission for. If empty, it defaults to Session user.
+     * @param string $categoryPermission The category permission to test against the user.
+     * @return bool Whether the user can view the discussion.
+     * @throws Exception
+     */
+    public function canView($discussion, $userID = '', $categoryPermission = 'Vanilla.Discussions.View') {
+        // Default to session user.
+        if (!$userID) {
+            $userID = val('UserID', Gdn::session(), false);
+        }
+        // Fetch discussion.
+        if (is_numeric($discussion)) {
+            $discussion = $this->getID($discussion);
+        }
+        $userModel = new UserModel();
+        // Get category permission.
+        $canView = $userID && $userModel->getCategoryViewPermission($userID, val('CategoryID', $discussion), $categoryPermission);
+
+        // Fire event to override permission ruling.
+        $this->EventArguments['Discussion'] = $discussion;
+        $this->EventArguments['UserID'] = $userID;
+        $this->EventArguments['CategoryPermission'] = $categoryPermission;
+        $this->EventArguments['CanView'] = &$canView;
+        $this->fireEvent('viewPermission');
```

Comment:
The argument is referencing the variable, not the method. Also, I agree with your feedback from a couple days ago:
>I'd rather the event arguments stay uppercase to be consistent with the rest of the framework. This isn't something we can easily change.
I think this should stay uppercase.